### PR TITLE
compiler warning: comparison of different signedness

### PR DESCRIPTION
**Pull request by jjsjann123**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -278,7 +278,7 @@ add_library(codegen_internal OBJECT ${NVFUSER_SRCS})
 if(NOT MSVC)
   target_compile_options(codegen_internal PRIVATE
     -Wall -Wno-unused-function
-    # -Werror
```
Comment (jjsjann123) [inline CMakeLists.txt:281]: I forgot why we turned off this.

**Comment by jjsjann123** [comment]
!build

**Review by jacobhinkle — APPROVED** [review]
Don't know why it was turned off either but thanks for catching this

File: CMakeLists.txt (hunk 1 above)
Comment (zasdfgbnm) [reply CMakeLists.txt:281]: The manylinux container is using gcc 12.2, which is not supported. This version of gcc have a bug that makes it unable to compile one of my PR with `-Werror`.

Comment (zasdfgbnm) [reply CMakeLists.txt:281]: Ideally, we should upgrade to gcc 12.4, but I tried that, and it is very hard compile gcc 12.4 in AlmaLinux 8....

Comment (jacobhinkle) [reply CMakeLists.txt:281]: Maybe we should just turn on `-Werror` for the CI clang-build job

Comment (jjsjann123) [reply CMakeLists.txt:281]: sounds good. I'll back out from that.

File: CMakeLists.txt [hunk 2]
```diff
@@ -278,6 +278,9 @@ endif()
 add_library(codegen_internal OBJECT ${NVFUSER_SRCS})
 
 if(NOT MSVC)
+  # -Werror is not enabled, because of gcc 12.2 used in manylinux image.
+  # consider enable this when we upgrad. linking comment:
+  # https://github.com/NVIDIA/Fuser/pull/3001#discussion_r1772551266
```
Comment (jjsjann123) [inline CMakeLists.txt:283]: revert the changes and added a comment on why we don't have it.

File: CMakeLists.txt [hunk 3]
```diff
@@ -278,6 +278,9 @@ endif()
 add_library(codegen_internal OBJECT ${NVFUSER_SRCS})
 
 if(NOT MSVC)
+  # -Werror is not enabled, because of gcc 12.2 used in manylinux image.
+  # consider enable this when we upgrad. linking comment:
```
Comment (jacobhinkle) [inline CMakeLists.txt:282]:
```suggestion
  # consider enable this when we upgrade. linking comment:
```

**Review by jacobhinkle — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (zasdfgbnm) [reply CMakeLists.txt:283]: BTW, because `-Wall` is already broken on gcc 12.2, we can continue the C++20 migration plan. Because the `-Wall` was the blocker, but now, `-Wall` is disabled due to a failure in C++17, we can just continue ignoring the warnings.

Comment (jacobhinkle) [reply CMakeLists.txt:283]: Reference: #3666
